app/main/views/frameworks.py

```diff
@@ -273,6 +273,73 @@ def framework_start_supplier_declaration(framework_slug):
                            framework_close_date=framework_close_date), 200
 
 
+@main.route('/frameworks/<framework_slug>/declaration/reuse', methods=['GET'])
+@login_required
+def reuse_framework_supplier_declaration(framework_slug):
+    """Attempt to find a supplier framework declaration that we can reuse.
+
+    :param: framework_slug
+    :query_param: reusable_declaration_framework_slug
+    :return: 404, redirect or reuse page (frameworks/reuse_declaration.html)
+    """
+    reusable_declaration_framework_slug = request.args.get('reusable_declaration_framework_slug', None)
+    supplier_id = current_user.supplier_id
+    # Get the current framework.
+    current_framework = data_api_client.get_framework(framework_slug)['frameworks']
+    if reusable_declaration_framework_slug:
+        # If they have supplied a declaration then force it. This is an 'admin' feature.
+        # Experimental feature to be used to force through reuse.
+        try:
+            # Get their old declaration.
+            declaration = data_api_client.get_supplier_declaration(supplier_id, reusable_declaration_framework_slug)
+            old_framework = data_api_client.get_framework(reusable_declaration_framework_slug)['frameworks']
+        except APIError:
+            abort(404)
+    else:
+        # Otherwise then attempt to determine if they have a declaration we can offer for reuse.
+        declarations = data_api_client.find_supplier_declarations(supplier_id)['frameworkInterest']
+        old_framework, declaration = get_reusable_declaration(
+            declarations,
+            client=data_api_client,
+            exclude_framework_slugs=[framework_slug]
+        )
+        if not declaration:
+            # If not then set the prefill to none and redirect to the overview.
+            data_api_client.set_supplier_framework_prefill_declaration(
+                supplier_id,
+                framework_slug,
+                None,
+                current_user.email_address
+            )
+            return redirect(url_for('.framework_supplier_declaration_overview', framework_slug=framework_slug))
+
+    # Otherwise offer to prefill the declaration.
+    return render_template(
+        "frameworks/reuse_declaration.html",
+        current_framework=current_framework,
+        old_framework=old_framework,
+        old_framework_application_close_date=date_parse(old_framework['application_close_date']).strftime('%B, %Y'),
+    ), 200
+
+
+@main.route('/frameworks/<framework_slug>/declaration/reuse', methods=['POST'])
+@login_required
+def reuse_framework_supplier_declaration_post(framework_slug):
+    try:
+        old_framework_slug = bool(int(request.form.get('reuse', 0))) or None
+    except ValueError:
+        old_framework_slug = request.form.get('reuse')
+
+    supplier_id = current_user.supplier_id
```

Comment:
I think we should add a check here that the supplier has a completed declaration for the specified framework and 404 if the declaration doesn't exist.  Would be weird to end up setting a value that says to reuse a declaration that we don't have.